cmdutil/debug/debug.go

```
@@ -71,3 +76,81 @@ func (s *Server) Stop(_ error) {
 
 	close(s.done)
 }
+
+// PProfServer wraps a pprof server.
+type PProfServer struct {
+	logger      logrus.FieldLogger
+	addr        string
+	done        chan struct{}
+	pprofServer *http.Server
+}
+
+// ProfileConfig holds the configuration for the pprof server.
+type PProfServerConfig struct {
+	Addr                 string
+	MutexProfileFraction int
+}
+
+// defaultMutexProfileFraction is the default value for MutexProfileFraction
+const defaultMutexProfileFraction = 2
+
+// NewPProfServer sets up a pprof server with configurable profiling types and returns a PProfServer instance.
+func NewPProfServer(config PProfServerConfig, l logrus.FieldLogger) *PProfServer {
+	if config.Addr == "" {
+		config.Addr = "127.0.0.1:9998" // Default port
+	}
+
+	// Use a local variable for the mutex profile fraction
+	mpf := defaultMutexProfileFraction
+	if config.MutexProfileFraction != 0 {
+		mpf = config.MutexProfileFraction
+	}
+	runtime.SetMutexProfileFraction(mpf)
+
+	httpServer := &http.Server{
+		Addr:              config.Addr,
+		Handler:           http.HandlerFunc(pprof.Index),
+		ReadHeaderTimeout: 5 * time.Second,
+	}
+
+	return &PProfServer{
+		logger:      l,
+		addr:        config.Addr,
+		done:        make(chan struct{}),
+		pprofServer: httpServer,
+	}
+}
+
+// Run starts the pprof server.
+//
+// It implements oklog group's runFn.
+func (s *PProfServer) Run() error {
+	s.logger.WithFields(logrus.Fields{
+		"at":      "binding",
+		"service": "pprof",
+		"addr":    s.addr,
+	}).Info()
+
+	if s.pprofServer != nil {
```

If it is nil, I think we can return an error.